public/app/plugins/datasource/loki/streaming.ts

```diff
@@ -86,3 +86,12 @@ export function doLokiChannelStream(
     })
   );
 }
+
+export const convertToWebSocketUrl = (url: string) => {
+  const protocol = window.location.protocol === 'https:' ? 'wss://' : 'ws://';
+  let backend = `${protocol}${window.location.host}${config.appSubUrl}`;
+  if (backend.endsWith('/')) {
+    backend = backend.slice(0, -1);
+  }
+  return `${backend}${url}`;
+};
```

Comment:
Should we add a small set of tests for this?

Reply:
Added in https://github.com/grafana/grafana/pull/78542/commits/ac2bf7ad8551f1114b7391b422d06c31dae0992e

Reply:
Thank you!